spring-boot/config-server/src/main/java/eu/coatrack/admin/server/service/filter/CheckIfUrlAndCredentialsFitWithEachOtherFilter.java

```diff
@@ -42,24 +42,19 @@
     private final String CREDENTIAL_URI = "/credentials";
 
     @Override
-    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
-        if (request instanceof HttpServletRequest
+    protected void doFilterInternal(HttpServletRequest servletRequest, HttpServletResponse servletResponse, FilterChain filterChain) throws ServletException, IOException {
+        if (!servletResponse.isCommitted()
                 // added the following check to avoid "IllegalStateException: resource already commited"
-                && !response.isCommitted()) {
-            /**
-             * Cast request/response
-             */
-            HttpServletRequest httpRequest = (HttpServletRequest) request;
-            HttpServletResponse httpResponse = (HttpServletResponse) response;
+                 ) {
 
             // Retrieve URI
-            String url = httpRequest.getRequestURI();
+            String url = servletRequest.getRequestURI();
 
             // Accepted URI - only check in case it is a proxy URI, other cases are covered by WebSecurityConfig
             if (url.startsWith(PROXY_URI)) { // || url.startsWith(ADMIN_URI) || url.startsWith(CREDENTIAL_URI)) {
 
                 // Retrieve the user/password
-                final String authorization = httpRequest.getHeader("Authorization");
+                final String authorization = servletRequest.getHeader("Authorization");
                 if (authorization != null && authorization.toLowerCase().startsWith("basic")) {
                     // Authorization: Basic base64credentials
```

Comment:
In general, Imo there are a lot of comments that do not add any value to the codes readability. Excessive need of comments is an indication for non-expressive code.